Content.Shared/_Harmony/Clothing/Gloves/Systems/SharedMantisGlovesSystem.cs

```diff
@@ -38,26 +40,39 @@ private void OnToggled(EntityUid uid, MantisGlovesComponent component, ref ItemT
 
         if (args.User != null)
         {
-            var message = args.Activated ? Loc.GetString("mantis-gloves-activated") : Loc.GetString("mantis-gloves-deactivated");
+            var message = args.Activated ? Loc.GetString(component.ActivatedPopUp!) : Loc.GetString(component.DeactivatedPopUp!);
             _popup.PopupClient(message, uid, args.User.Value);
         }
     }
 
     /// <summary>
     /// Updates the metadata (name and description) of mantis gloves based on their activation state.
+    /// When activated, uses the localization keys from the component
+    /// When deactivated, uses the prototype values but wraps them in Loc.GetString
     /// </summary>
     private void UpdateMantisGlovesState(EntityUid uid, bool activated, MantisGlovesComponent? component = null)
     {
         if (!Resolve(uid, ref component))
             return;
 
-        var name = activated ? component.ActivatedName : component.DeactivatedName;
-        var description = activated ? component.ActivatedDescription : component.DeactivatedDescription;
+        var meta = MetaData(uid);
+        var protoId = meta.EntityPrototype?.ID;
 
-        if (name != null)
-            _metaData.SetEntityName(uid, Loc.GetString(name));
+        if (protoId == null)
+            return;
+
+        if (!_prototypeManager.TryIndex<EntityPrototype>(protoId, out var prototype))
+            return;
+
+        string name = activated
+            ? Loc.GetString(component.ActivatedName!)
+            : prototype.Name;
+
+        string description = activated
+            ? Loc.GetString(component.ActivatedDescription!)
+            : prototype.Description;
```

Comment:
In theory, we can avoid all of this and just put it like so
```cs
string name = activated ? component.ActivatedName : prototype.Name

_metaData.SetEntityDescription(uid, Loc.GetString(description));
```
Since `Loc.GetString` handels the possibility of said string not existing in `.ftl`, but I don't think this is a good practice